### PR TITLE
Chore/duplicate-pages-in-api

### PR DESCRIPTION
**Pull request by dewm3734**
Added a catch clause so that if a "skip_docs" methods is present within the individual generation files then those files will not be generated

### CODE REVIEW

**Review by ellerynz — COMMENTED** [review]
Seems good to me.
- Can you fix the rubocop issue?
- And can you try use this branch locally in Investapp? Then we'll know that it's behaving like we expect

**Comment by dewm3734** [comment]
@ellerynz the error is `test/examples/exercise12_skipped.rb:36:4: C: Layout/TrailingEmptyLines: Final newline missing.` I am not too sure what this means. I have managed to get it working locally and it works perfectly,

File: test/examples/exercise12_skipped.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        field :v2_0_only, version: { equals: 'v2.0.*-*' }
+        field :v2_1_only, version: { equals: 'v2.1.*-*' }
+
+        field :params
+        field :options
+      end
+    end
+  end
+end
```
Comment (ellerynz) [inline test/examples/exercise12_skipped.rb:36]: @dewm3734 it looks like Rubocop is picking up the lack of a newline at the end of the file. The fix is to add a newline here and if you're interested in some of the history, give [this a read](https://thoughtbot.com/blog/no-newline-at-end-of-file).

**Review by OBaddeley — COMMENTED** [review]
Nice 👍🏼 
- Also wondering if you have been able to run this locally with investapp to check it is working as expected?

**Comment by dewm3734** [comment]
@OBaddeley Yip this runs locally and is being tested on the investapp side

**Pull request merged**